real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/logging/SensorLogger.java

```diff
@@ -144,36 +173,37 @@ private static String getCurrentSoftwareVersion() {
 		return "1.0.0";
 	}
 
-	SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss");
-
-	public SensorLogger(File outputDir, Date startTime) throws Exception {
+	 /**Construct a new {@link SensorLogger} object
+	  **@param outputDir {@link File} of the output file directory
+	  */
+	public SensorLogger(File outputDir) {
 		if (outputDir == null) {
-			throw new Exception("Output Directory was null!");
+			throw new IllegalArgumentException("Output Directory was null!");
 		} else if (!outputDir.exists()) {
-			outputDir.mkdirs();
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1892352955/issues/source?bid=2938665&fileBranchId=3087231#l183)